client/db/src/storage_cache.rs

```diff
@@ -852,7 +861,11 @@ impl<S: StateBackend<HashFor<B>>, B: BlockT> StateBackend<HashFor<B>> for Syncin
 			.as_trie_backend()
 	}
 
-	fn usage_info(&self) -> sp_state_machine::UsageInfo {
+	fn register_overlay_stats(&mut self, stats: &sp_stats::StateMachineStats) {
```

Comment:
the signature/name of this function must somehow ensure that it consumes stats (i.e. we never add something twice

Reply:
Yes so as explained in the PR, I've merged Cheme's WIP PR. I think it adds a lot of confusion to my PR, but it kind of touches the same areas of the code...
I think I can safely revert this merge. Will simply running `git revert -m 1 COMMIT` work, even if we merge his PR later on?

Reply:
I reverted the merge